src/axom/quest/examples/shaping_driver.cpp

```diff
@@ -223,6 +224,28 @@ struct Input
       ->transform(
         axom::CLI::CheckedTransformer(methodMap, axom::CLI::ignore_case));
 
+#ifdef AXOM_USE_CALIPER
+    app.add_option("--caliper", annotationMode)
+      ->description(
+        "caliper annotation mode. Valid options include 'none' and 'report'. "
+        "Use 'help' to see full list.")
+      ->capture_default_str()
+      ->check([](const std::string& mode) -> std::string {
+        if(mode == "help")
+        {
+          std::cerr << "Valid caliper modes are:\n"
+                    << axom::utilities::annotations::detail::mode_help_string()
+                    << std::endl;
+        }
+        return axom::utilities::annotations::detail::is_mode_valid(mode)
+          ? ""
+          : fmt::format(
+              "'{}' invalid caliper mode. "
+              "Run with '--caliper help' to see all valid options",
+              mode);
+      });
+#endif
```

Comment:
Consider encapsulating this 20-line formula somewhere.  It's copy-pasted three places so far in this PR.  That's not fatal, especially since the uses are all independent and serve as example code, but this way to collect and check Caliper options is non-trivial to get just right.

Reply:
Good call -- The lambda can be converted to a CLI11 Validator, which should save about 10 lines (233--245 here).

Reply:
I'll figure out how to encapsulate this better in a follow-up PR.

Tracked in https://github.com/LLNL/axom/issues/1314